src/rf/apps/workers/process.py

```diff
@@ -0,0 +1,221 @@
+
+        # TODO - Right now we don't save data to the DB with upload. So hard
+        # code something for testing.
+        try:
+            image = LayerImage.objects.get(s3_uuid=payload_uuid)
+            image.status = STATUS_VALIDATED
+            image.save()
+        except:
+            payload_uuid = '1aa064aa-1086-4ff1-a90b-09d3420e0343'
+
+        data = {'url': url, 's3_uuid': payload_uuid}
+        payload = self.make_payload(JOB_REPROJECT, data)
+        self.post_to_queue(payload)
+        return True
+
+    def reproject(self, data):
+        """
+        Reproject incoming image to Web Mercator.
+        data -- attribute data from SQS.
+        """
+        # Pass image to Gdal to reproject
+        # If it succeeds continue else return False.
+
+        s3_uuid = data['s3_uuid']
+        image = LayerImage.objects.get(s3_uuid=s3_uuid)
+        image.status = STATUS_REPROJECTED
+        image.save()
+
+        data = {'layer_id': image.layer_id}
+        payload = self.make_payload(JOB_HANDOFF, data)
+        self.post_to_queue(payload)
+        return True
+
+    def emr_hand_off(self, data):
+        """
+        Passes layer imgages to EMR to begin creating custom rasters.
+        data -- attribute data from SQS.
+        """
+        # If all the images are ready send data to EMR for processing.
+        # return False if it fails to start.
+        # EMR posts directly to queue upon competion.
+
+        layer_id = data['layer_id']
+        layer_images = LayerImage.objects.filter(layer_id=layer_id)
+        reprojected_images = LayerImage.objects.filter(layer_id=layer_id,
+                                                       status=STATUS_REPROJECTED)  # NOQA
+        ready_to_process = len(layer_images) == len(reprojected_images)
+
+        if ready_to_process:
+            layer_meta = LayerMeta.objects.get(layer_id=layer_id)
+            layer_meta.state = STATUS_PROCESSING
+            layer_meta.save()
+
+            # POST TO EMR HERE.
+
+            # Add a message to the queue that we can use to watch for
+            # EMR timeout.
+            data = {
+                'timeout': time.time() + TIMEOUT_SECONDS,
+                'layer_id': layer_id
+            }
+
+            payload = self.make_payload(JOB_TIMEOUT, data)
+            self.post_to_queue(payload)
+
+        return True
+
+    def check_timeout(self, data):
+        """
+        Check an EMR job to see if it has completed before the timeout has
+        been reached.
+        data -- attribute data from SQS.
+        """
+
+        layer_id = data['layer_id']
+        timeout = data['timeout']
+        if time.time() > timeout:
+            layer_meta = LayerMeta.objects.get(layer_id=layer_id)
+            if not layer_meta.state == STATUS_COMPLETE:
+                layer_meta.state = STATUS_FAILED
+                layer_meta.save()
+            return True
+        else:
+            return False
```

Comment:
Instead of returning False, we should return True to delete the current message and enqueue a new message. Otherwise, this could be sent to the dead letter queue if we fail to process it a certain number of times before the timeout limit is reached.